### ISSUE TITLE
Do i need instance_id for training VID model?

### ISSUE THREAD

**Issue by Cospel** [open-mmlab/mmtracking#170]
Hi, I'm trying to train the custom model with several categories for Video object detection.

**Do I need to have 'instance_id' for the video object detection problem?** As I read here the instance_id should be required only for tracking:
https://github.com/open-mmlab/mmtracking/blob/ecc83b5e6523582b92196095eb21d72d654322f2/docs/tutorials/customize_dataset.md

However, in configurations for the SELSA model in configuration imagenet_vid_fgfa_style.py is  with_track=True which is working with instance_id.

Right now I'm unable to get instance_ids, so I put to every annotation unique number. The model accuracy during training shows more than 96 %, and after 100 iterations the loss value is pretty much flattened. When testing the model on real video (from training), the performance is very low.

So the questions are:
1) do I need instance_id for training vid models? or can this be unique integer values per annotation ...
2) is it possible to compute mAP?

    ```
loading annotations into memory...
    Done (t=0.25s)
    creating index...
    index created!
    2021-06-03 14:14:30,800 - mmtrack - INFO - Start running, host: michal@compute-04, work_dir: /home/michal/dev/mmtracking/log
    2021-06-03 14:14:30,801 - mmtrack - INFO - workflow: [('train', 1)], max: 7 epochs
    Group size [8093]
    /home/michal/dev/lib64/python3.6/site-packages/mmdet/models/dense_heads/rpn_head.py:192: UserWarning: In rpn_proposal or test_cfg, nms_thr has been moved to a dict named nms as iou_threshold, max_num has been renamed as max_per_img, name of original arguments and the way to specify iou_threshold of NMS will be deprecated.
      'In rpn_proposal or test_cfg, '
    2021-06-03 14:15:43,890 - mmtrack - INFO - Epoch [1][50/8093]	lr: 1.485e-04, eta: 22:58:12, time: 1.461, data_time: 0.066, memory: 3453, loss_rpn_cls: 0.3530, loss_rpn_bbox: 0.0405, loss_cls: 0.7523, acc: 90.4617, loss_bbox: 0.0300, loss: 1.1758, grad_norm: 3.0899
    2021-06-03 14:16:54,425 - mmtrack - INFO - Epoch [1][100/8093]	lr: 2.970e-04, eta: 22:33:19, time: 1.411, data_time: 0.018, memory: 3453, loss_rpn_cls: 0.1960, loss_rpn_bbox: 0.0486, loss_cls: 0.1531, acc: 98.4218, loss_bbox: 0.0267, loss: 0.4244, grad_norm: 2.6038
    2021-06-03 14:18:04,220 - mmtrack - INFO - Epoch [1][150/8093]	lr: 3.000e-04, eta: 22:19:33, time: 1.396, data_time: 0.020, memory: 3453, loss_rpn_cls: 0.1584, loss_rpn_bbox: 0.0364, loss_cls: 0.1658, acc: 97.9499, loss_bbox: 0.0331, loss: 0.3937, grad_norm: 2.3144

```

**Comment by GT9505** [comment]
1. No. Currently,  all video detection methods in mmtracking don't need annotations of instance ids.  We set `with_track=True` only for providing interfaces in case of reimplementing video detection methods which need instance id. You can set `with_track=False` in SELSA.
2. mmtracking has already supported to report mAP after each epoch.
3. As for the low performance, do you mean that you test on the training video?

**Comment by Cospel** [comment]
Thank you for your response! Really appreciate it.

1. You are right, I must also comment the gt_instance_ids when with_track = False
2. That is kind of strange, I'm using the configs provided from **SELSA**, however cannot see any map results after the epoch is done. Do I need to somehow configure it?
3. I still try to tune hyperparams and probably frame rate is important.

**Comment by GT9505** [comment]
2. You can change the value of `interval` in [here](https://github.com/open-mmlab/mmtracking/blob/master/configs/vid/selsa/selsa_faster_rcnn_r50_dc5_1x_imagenetvid.py#L48). It means we validate the trained model every `interval` epochs.

**Comment by Cospel** [comment]
Thank you,
one last thing about loss function for vid problem. Do you know why the loss functions are not improving? After the first 50 steps, the loss values are just oscillating forever ... I'm not sure why is this happening, I use your default configurations for the training of SELSA vid model.


2021-06-08 08:57:50,008 - mmtrack - INFO - Epoch [1][50/8777]   lr: 3.987e-03, eta: 1 day, 9:08:16, time: 1.943, data_time: 0.061, memory: 4755, loss_rpn_cls: 0.4379, loss_rpn_bbox: 0.0274, loss_cls: 0.9259, acc: 90.0007, loss_bbox: 0.0183, loss: 1.4095, grad_norm: 2.2423
2021-06-08 08:59:25,385 - mmtrack - INFO - Epoch [1][100/8777]  lr: 4.653e-03, eta: 1 day, 8:48:20, time: 1.907, data_time: 0.013, memory: 4755, loss_rpn_cls: 0.1276, loss_rpn_bbox: 0.0540, loss_cls: 0.1054, acc: 98.7021, loss_bbox: 0.0083, loss: 0.2953, grad_norm: 1.2635
2021-06-08 09:01:00,624 - mmtrack - INFO - Epoch [1][150/8777]  lr: 5.320e-03, eta: 1 day, 8:39:42, time: 1.905, data_time: 0.014, memory: 4755, loss_rpn_cls: 0.1427, loss_rpn_bbox: 0.0890, loss_cls: 0.1212, acc: 98.4170, loss_bbox: 0.0147, loss: 0.3675, grad_norm: 1.4783
2021-06-08 09:02:35,425 - mmtrack - INFO - Epoch [1][200/8777]  lr: 5.987e-03, eta: 1 day, 8:32:21, time: 1.896, data_time: 0.014, memory: 4755, loss_rpn_cls: 0.1620, loss_rpn_bbox: 0.0608, loss_cls: 0.1835, acc: 97.8175, loss_bbox: 0.0225, loss: 0.4287, grad_norm: 1.4336
2021-06-08 09:04:09,604 - mmtrack - INFO - Epoch [1][250/8777]  lr: 6.653e-03, eta: 1 day, 8:24:47, time: 1.884, data_time: 0.015, memory: 4755, loss_rpn_cls: 0.1834, loss_rpn_bbox: 0.0850, loss_cls: 0.2874, acc: 95.6014, loss_bbox: 0.0481, loss: 0.6038, grad_norm: 1.6621
2021-06-08 09:05:44,298 - mmtrack - INFO - Epoch [1][300/8777]  lr: 7.320e-03, eta: 1 day, 8:20:58, time: 1.894, data_time: 0.013, memory: 4755, loss_rpn_cls: 0.1743, loss_rpn_bbox: 0.1160, loss_cls: 0.1533, acc: 97.8796, loss_bbox: 0.0225, loss: 0.4661, grad_norm: 1.3940
2021-06-08 09:07:18,990 - mmtrack - INFO - Epoch [1][350/8777]  lr: 7.987e-03, eta: 1 day, 8:17:47, time: 1.894, data_time: 0.013, memory: 4755, loss_rpn_cls: 0.1660, loss_rpn_bbox: 0.0456, loss_cls: 0.2294, acc: 96.5558, loss_bbox: 0.0446, loss: 0.4856, grad_norm: 1.3279
2021-06-08 09:08:52,829 - mmtrack - INFO - Epoch [1][400/8777]  lr: 8.653e-03, eta: 1 day, 8:12:50, time: 1.877, data_time: 0.011, memory: 4755, loss_rpn_cls: 0.1581, loss_rpn_bbox: 0.0297, loss_cls: 0.2869, acc: 95.4847, loss_bbox: 0.0561, loss: 0.5308, grad_norm: 1.4185
2021-06-08 09:10:28,491 - mmtrack - INFO - Epoch [1][450/8777]  lr: 9.320e-03, eta: 1 day, 8:12:44, time: 1.913, data_time: 0.015, memory: 4755, loss_rpn_cls: 0.1483, loss_rpn_bbox: 0.0511, loss_cls: 0.1745, acc: 97.6879, loss_bbox: 0.0296, loss: 0.4035, grad_norm: 1.0784
2021-06-08 09:12:02,623 - mmtrack - INFO - Epoch [1][500/8777]  lr: 9.987e-03, eta: 1 day, 8:09:15, time: 1.883, data_time: 0.015, memory: 4755, loss_rpn_cls: 0.1639, loss_rpn_bbox: 0.0773, loss_cls: 0.1745, acc: 97.4950, loss_bbox: 0.0197, loss: 0.4354, grad_norm: 0.8977
2021-06-08 09:13:36,384 - mmtrack - INFO - Epoch [1][550/8777]  lr: 1.000e-02, eta: 1 day, 8:05:25, time: 1.875, data_time: 0.013, memory: 4755, loss_rpn_cls: 0.1427, loss_rpn_bbox: 0.0311, loss_cls: 0.2059, acc: 97.1131, loss_bbox: 0.0285, loss: 0.4082, grad_norm: 0.9178
2021-06-08 09:15:10,603 - mmtrack - INFO - Epoch [1][600/8777]  lr: 1.000e-02, eta: 1 day, 8:02:45, time: 1.884, data_time: 0.015, memory: 4755, loss_rpn_cls: 0.1407, loss_rpn_bbox: 0.0337, loss_cls: 0.1886, acc: 96.9944, loss_bbox: 0.0334, loss: 0.3964, grad_norm: 0.8163
2021-06-08 09:16:44,794 - mmtrack - INFO - Epoch [1][650/8777]  lr: 1.000e-02, eta: 1 day, 8:00:12, time: 1.884, data_time: 0.016, memory: 4755, loss_rpn_cls: 0.1268, loss_rpn_bbox: 0.0363, loss_cls: 0.1737, acc: 97.3153, loss_bbox: 0.0277, loss: 0.3646, grad_norm: 0.7395
2021-06-08 09:18:20,880 - mmtrack - INFO - Epoch [1][700/8777]  lr: 1.000e-02, eta: 1 day, 8:00:31, time: 1.922, data_time: 0.018, memory: 4755, loss_rpn_cls: 0.1456, loss_rpn_bbox: 0.0443, loss_cls: 0.1314, acc: 98.1041, loss_bbox: 0.0184, loss: 0.3397, grad_norm: 0.8541
2021-06-08 09:19:55,551 - mmtrack - INFO - Epoch [1][750/8777]  lr: 1.000e-02, eta: 1 day, 7:58:42, time: 1.894, data_time: 0.014, memory: 4755, loss_rpn_cls: 0.1519, loss_rpn_bbox: 0.0374, loss_cls: 0.1354, acc: 98.0198, loss_bbox: 0.0152, loss: 0.3399, grad_norm: 0.8846
2021-06-08 09:21:31,498 - mmtrack - INFO - Epoch [1][800/8777]  lr: 1.000e-02, eta: 1 day, 7:58:30, time: 1.919, data_time: 0.016, memory: 4755, loss_rpn_cls: 0.1719, loss_rpn_bbox: 0.0416, loss_cls: 0.1639, acc: 97.5833, loss_bbox: 0.0279, loss: 0.4054, grad_norm: 0.8152
2021-06-08 09:23:08,465 - mmtrack - INFO - Epoch [1][850/8777]  lr: 1.000e-02, eta: 1 day, 7:59:22, time: 1.939, data_time: 0.015, memory: 4755, loss_rpn_cls: 0.1371, loss_rpn_bbox: 0.0325, loss_cls: 0.1211, acc: 98.3909, loss_bbox: 0.0166, loss: 0.3072, grad_norm: 0.8180
2021-06-08 09:24:43,937 - mmtrack - INFO - Epoch [1][900/8777]  lr: 1.000e-02, eta: 1 day, 7:58:16, time: 1.909, data_time: 0.015, memory: 4755, loss_rpn_cls: 0.1648, loss_rpn_bbox: 0.0454, loss_cls: 0.1940, acc: 97.0784, loss_bbox: 0.0355, loss: 0.4397, grad_norm: 1.0137
2021-06-08 09:26:18,365 - mmtrack - INFO - Epoch [1][950/8777]  lr: 1.000e-02, eta: 1 day, 7:56:00, time: 1.888, data_time: 0.016, memory: 4755, loss_rpn_cls: 0.1803, loss_rpn_bbox: 0.0349, loss_cls: 0.2484, acc: 96.1422, loss_bbox: 0.0387, loss: 0.5023, grad_norm: 1.0698
2021-06-08 09:27:53,109 - mmtrack - INFO - Saving checkpoint at 1000 iterations
2021-06-08 09:27:54,499 - mmtrack - INFO - Epoch [1][1000/8777] lr: 1.000e-02, eta: 1 day, 7:55:33, time: 1.923, data_time: 0.016, memory: 4755, loss_rpn_cls: 0.1548, loss_rpn_bbox: 0.0384, loss_cls: 0.2104, acc: 96.5273, loss_bbox: 0.0434, loss: 0.4469, grad_norm: 1.0270

**Comment by GT9505** [comment]
You can try to train the model a little longer.
The [log](https://download.openmmlab.com/mmtracking/vid/selsa/selsa_faster_rcnn_r101_dc5_1x_imagenetvid/selsa_faster_rcnn_r101_dc5_1x_imagenetvid_20201218_172724.log.json) of training SELSA on ImageNet VID dataset shows similar phenomena with you in the first 1000 iterations.

**Comment by Cospel** [comment]
Thank you, you can close this.

**Comment by lihaoliu-cambridge** [comment]
> Thank you for your response! Really appreciate it.
> 
> 1. You are right, I must also comment the gt_instance_ids when with_track = False
> 2. That is kind of strange, I'm using the configs provided from **SELSA**, however cannot see any map results after the epoch is done. **Do I need to somehow configure it?**
> 3. I still try to tune hyperparams and probably frame rate is important.

Hi, Michal Lukac. Hope you are doing great. I meet the same problem as you mentioned here, with my own dataset without instance_id. When I train it, the loss is decreasing, and acc is ~95%. But when I test, all mAPs are -1, which is clearly wrong. I am wondering is there any steps I missed to run on customed dataset?

(1). I set "with_track=False" in the config files.
(2). I delete "gt_instance_ids" in the config files.
(3). I add class information for my customer dataset in config files.
(4). I change the "num_classes" in the model config files.

or it is something wrong with the testing stage?

<img width="1463" alt="Screenshot 2023-02-08 at 01 03 02" src="https://user-images.githubusercontent.com/19239227/217402246-05f7fbf1-be46-4551-8697-e5bbbcfec21e.png">
